### DISCUSSION TITLE
Scaffolding starter templates with Remix

### DISCUSSION THREAD

**Discussion by ShafSpecs** [remix-run/remix#1472] [Proposals]
By scaffolding starter templates, I mean remix would have some preset templates, plain but functional that would allow users to either kickstart a project or allow users to test something.

I feel this is something Remix needs as not everyone can, and want to start from a blank template. For exmple, I want to play around with loaders, I have to create an app with routes and fill it with as much stuff I need to make the site look more like smthg that would use a loader. Sometimes, I want to create an e-commerce and style with tailwind, would be awesome to just get started by scaffolding a plain ecommerce layout that's already connected to stripe and maybe even uses strapi to handle the e-commerce side of things. What I am saying in short is that the ability to build more on what we have at insallation time is very needed.

So, the way I think this can work is by having additional flags to support the `npx create-remix@latest` command. We would have starter templates that have been bootstrapped based on the user choice (e.g Tailwind, Supabase, Strapi, Socket.io, etc.)

**Comment by hollandThomas** [comment]
Maintaining all of this in Remix core would lead to lots of overhead I assume.

Instead, have a look at the examples folder of the repo
https://github.com/remix-run/remix/tree/main/examples

Or this website for more complete examples
https://remix.guide/

Both will only keep on growing in both quantity and quality. This might be what you are looking for
